src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+    const identity_path = default_config_fs.get_identity_path_by_id(_id);
+    const exists = await is_path_exists(default_config_fs.fs_context, access_key_path);
+    expect(exists).toBe(true);
+    const access_key_linked_to_identity = await default_config_fs._is_symlink_pointing_to_identity(access_key_path, identity_path);
+    expect(access_key_linked_to_identity).toBe(true);
+}
+
+/**
+ * assert_name_index_is_updated to point to the account's identity.json file
+ * @param {Object} account_data
+ * @returns {Promise<Void>}
+ */
+async function assert_name_index_is_updated(account_data) {
+    const { _id, name } = account_data;
+    const account_by_name_path = default_config_fs.get_account_or_user_path_by_name(name);
+    const identity_path = default_config_fs.get_identity_path_by_id(_id);
+    const exists = await is_path_exists(default_config_fs.fs_context, account_by_name_path);
+    expect(exists).toBe(true);
+    const name_linked_to_identity = await default_config_fs._is_symlink_pointing_to_identity(account_by_name_path, identity_path);
+    expect(name_linked_to_identity).toBe(true);
+}
+
+/**
+ * assert_identity_created asserts that -
+ * 1. the identity dir was created
+ * 2. the identity.json file was created
+ * @param {Object} account_data
+ * @returns {Promise<Void>}
+ */
+async function assert_identity_created(account_data) {
+    const { _id } = account_data;
+    const identity_dir_path = default_config_fs.get_identity_path_by_id(_id);
+    const dir_exists = await is_path_exists(default_config_fs.fs_context, identity_dir_path);
+    expect(dir_exists).toBe(true);
+    const identity_data = await default_config_fs.get_identity_by_id(_id, CONFIG_TYPES.ACCOUNT, { show_secrets: true });
+    expect(identity_data).toStrictEqual(account_data);
+}
+
+/**
+ * assert_upgrade_params asserts that the actual upgrade params are as expected
+ * @param {Object} actual_upgrade_params 
+ * @param {Object} expected_account_upgrade_params
+ * @returns {Void} 
+ */
+function assert_upgrade_params(actual_upgrade_params, expected_account_upgrade_params) {
+    expect(actual_upgrade_params).toStrictEqual(expected_account_upgrade_params);
+}
+
+/**
+ * assert_account_config_file_upgraded asserts that the account config file was upgraded as expected
+ */
+async function assert_account_config_file_upgraded(expected_accounts) {
+    for (const account_name of Object.keys(expected_accounts)) {
+        const account_data = expected_accounts[account_name];
+        await assert_access_key_index_is_updated(account_data);
+        await assert_name_index_is_updated(account_data);
+        await assert_identity_created(account_data);
+    }
+}
+
+// Jest has builtin function fail that based on Jasmine
+// in case Jasmine would get removed from jest, created this one
+// based on this: https://stackoverflow.com/a/55526098/16571658
+function fail(reason) {
+    throw new Error(reason);
+}
```

Comment:
If you want we can move it to a new file "jest_utils" (not necessarily in this PR).

Reply:
I wanted to do it but it's being re-used in many many files and this PR is already very large, please add it to the unit tests refactoring issue